int8 num
int8 mode
float64 x
float64 y
float64 z
float64 theta_x
float64 theta_y
float64 theta_z
